### PR TITLE
Properly check and guard overloading initializer name | fix(graph_building)

### PR DESCRIPTION
**Pull request by BowenBao**
Previously the code path is never triggered due to another bug in exporter that initializers are always added with unique names.

### CODE REVIEW

**Comment by BowenBao** [comment]
Not sure which topic label to pick

File: onnxscript/function_libs/torch_lib/graph_building_test.py [hunk 1]
```diff
@@ -118,6 +118,18 @@ def outer(x: FLOAT[1, 2, 3]):
         expected = outer.to_model_proto()
         onnxscript.testing.assert_isomorphic(traced, expected)
 
+    def test_adding_different_tensors_using_same_initializer_name_raises(self):
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/graph_building_test.py:121]:
```suggestion
class TestTorchScriptGraph(unittest.TestCase):
    def test_add_initializer_raises_when_the_same_name_used_for_different_tensors(self):
    
```

Create a test class for the graph class and update the same to something along this line?

**Review by titaiwangms — APPROVED** [review]
I guess there should be another fix in PyTorch. Would you link this PR to the fixed in PyTorch?

File: onnxscript/function_libs/torch_lib/graph_building_test.py [hunk 2]
```diff
@@ -118,6 +118,18 @@ def outer(x: FLOAT[1, 2, 3]):
...
+        graph = graph_building.TorchScriptGraph()
+        graph.add_initializer("x", torch.ones((1, 2, 3), dtype=torch.float32))
+        with self.assertRaises(ValueError):
+            graph.add_initializer("x", torch.ones((1, 2, 3), dtype=torch.float32))
+
+    def test_adding_same_tensors_using_same_initializer_name_passes(self):
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/graph_building_test.py:127]:
```suggestion
    def test_add_initializer_allows_adding_the_same_tensor_twice_using_same_name(self):
```

etc.

File: onnxscript/function_libs/torch_lib/graph_building.py [hunk 3]
```diff
@@ -404,7 +404,7 @@ def add_initializer(self, name: str, value: torch.Tensor) -> TorchScriptTensor:
             # NOTE: Previously it raises when `name` is already set. This is relaxed
             # because this will be invoked multiple times when submodule is called
             # multiple times.
-            if name in self._initializers and self._initializers[name] != value:
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/graph_building.py:407]: So this comparison was invalid right?

Comment (BowenBao) [reply onnxscript/function_libs/torch_lib/graph_building.py:407]: It does tensor value comparison, i.e. torch.neq(a, b). It returns a bool tensor.

The check should instead see if they are same tensor object.

**Comment by BowenBao** [comment]
> I guess there should be another fix in PyTorch. Would you link this PR to the fixed in PyTorch?

No code changes on torch side. We'll bump commit in CI in next related PR.
